tree/tree/src/TTree.cxx

```diff
@@ -5581,11 +5581,13 @@ Long64_t TTree::GetEntriesFriend() const
 
 Int_t TTree::GetEntry(Long64_t entry, Int_t getall)
 {
-
    // We already have been visited while recursively looking
    // through the friends tree, let return
    if (kGetEntry & fFriendLockStatus) return 0;
-
+   if (!entry) {
```

Comment:
zero is a valid value!!!  It is the 'first' entry per se.

Reply:
But doesn't `!entry` mean that no value for entry was provided?

Reply:
Nope.  The user can provide 0 explicitly and there is no way of seeing the difference.  And since 0 is the first entry, it will be almost always being passed the first time `GetEntry` is called :)